### PR TITLE
react-form-controls - Justin Arakaki

### PR DESCRIPTION
**Pull request by Justin-Arakaki**
Not sure how they only console.log'd when the input field had a complete email but here is an approximation. It uses a timer that resets. The timer ID is stored in the state. It seemed like it should be part of the NewsletterForm class somewhere.
react-form-controls 1

### CODE REVIEW

File: react-form-controls/src/newsletter-form.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+    this.logState();
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
```
Comment (scbowler) [inline react-form-controls/src/newsletter-form.jsx:17]: You just needed to console log the state here, this method is called once the form is submitted.

File: react-form-controls/src/newsletter-form.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  logState() {
+    clearTimeout(this.state.timerID);
+    const timer = setTimeout(() => {
+      console.log('state:', this.state);
+    }, 1000);
+    this.setState({ timerID: timer });
+  }
```
Comment (scbowler) [inline react-form-controls/src/newsletter-form.jsx:26]: Creative but not the right way to go

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just need to update logging the state onSubmit. Let me know if you have any questions.

**Comment by Justin-Arakaki** [comment]
image

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**